samtranslator/model/sam_resources.py

```diff
@@ -2144,26 +2147,76 @@ class SamGraphQLApi(SamResourceMacro):
     Name: Optional[str]
     SchemaInline: Optional[str]
     SchemaUri: Optional[str]
+    Logging: Optional[Union[Dict[str, Any], bool]]
 
     @cw_timer
     def to_cloudformation(self, **kwargs: Any) -> List[Resource]:
-        appsync_api = self._construct_appsync_api()
+        appsync_api, cloudwatch_role = self._construct_appsync_api_resources()
         appsync_schema = self._construct_appsync_schema(appsync_api.get_runtime_attr("api_id"))
+
         resources: List[Resource] = [appsync_api, appsync_schema]
 
+        if cloudwatch_role:
+            resources.append(cloudwatch_role)
+
         return resources
 
-    def _construct_appsync_api(self) -> GraphQLApi:
+    def _construct_appsync_api_resources(self) -> Tuple[GraphQLApi, Optional[IAMRole]]:
         api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
 
         api.AuthenticationType = sam_expect(self.Auth.get("Type"), self.logical_id, "Auth.Type").to_be_a_string()
         api.Name = self.Name if self.Name else self.logical_id
         api.XrayEnabled = self.XrayEnabled
 
+        cloudwatch_role = None  # conditionally generated, if left as None then its not added to resources
```

Comment:
nit: this way of declaring has a couple of disadvantages
- type is missing
- it sucks to add type manually and then to get it return from the function anyway
- more difficult to read because actual assignment happens later

Suggestion: instead of declaring it here, do it when you return it from the function like this
```python
should_construct_new_role = isinstance(self.Logging, dict) and  "CloudWatchLogsRoleArn" not in self.Logging
cloudwatch_role = self._construct_cloudwatch_role() if should_construct_new_role else None
```